docs/topics/coroutines-basics.md

```diff
@@ -1,293 +1,565 @@
+Add the following dependency to your `build.gradle(.kts)` file:
+
+<tabs group="build-tool">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+// build.gradle.kts
+repositories {
+    mavenCentral()
+}
 
-[delay] is a special _suspending function_. It _suspends_ the coroutine for a specific time. Suspending a coroutine
-does not _block_ the underlying thread, but allows other coroutines to run and use the underlying thread for 
-their code. 
+dependencies {
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:%coroutinesVersion%")
+}
+```
 
-[runBlocking] is also a coroutine builder that bridges the non-coroutine world of a regular `fun main()` and 
-the code with coroutines inside of `runBlocking { ... }` curly braces. This is highlighted in an IDE by
-`this: CoroutineScope` hint right after the `runBlocking` opening curly brace.
+</tab>
+<tab title="Groovy" group-key="groovy">
 
-If you remove or forget `runBlocking` in this code, you'll get an error on the [launch] call, since `launch`
-is declared only on the [CoroutineScope]:
+```groovy
+// build.gradle
+repositories {
+    mavenCentral()
+}
 
+dependencies {
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:%coroutinesVersion%'
+}
 ```
-Unresolved reference: launch
+</tab>
+</tabs>
+
+### Maven
+
+Add the following dependency to your `pom.xml` file.
+
+```xml
+<project>
+    <dependencies>
+        <dependency>
+            <groupId>org.jetbrains.kotlinx</groupId>
+            <artifactId>kotlinx-coroutines-core</artifactId>
+            <version>%coroutinesVersion%</version>
+        </dependency>
+    </dependencies>
+    ...
+</project>
 ```
 
-The name of `runBlocking` means that the thread that runs it (in this case &mdash; the main thread) gets _blocked_ for 
-the duration of the call, until all the coroutines inside `runBlocking { ... }` complete their execution. You will 
-often see `runBlocking` used like that at the very top-level of the application and quite rarely inside the real code, 
-as threads are expensive resources and blocking them is inefficient and is often not desired. 
+## Create your first coroutines
+
+A coroutine is a suspendable computation that can run concurrently with other coroutines and potentially in parallel.
+
+On the JVM and in Kotlin/Native, all concurrent code, such as coroutines, runs on _threads_, which are managed by the operating system.
+Coroutines can suspend their execution instead of blocking a thread.
+This allows one coroutine to suspend while waiting for some data to arrive and another coroutine to run on the same thread, ensuring effective resource utilization.
+
+![Comparing parallel and concurrent threads](parallelism-and-concurrency.svg){width="700"}
+
+To create a coroutine in Kotlin, you need the following:
+
+* A suspending function.
+* A coroutine scope in which it can run, such as one available inside the `withContext()` function.
+* A coroutine builder like `.launch()` to start it.
```

Comment:
Here (and across the entire doc), let's use the full form `CoroutineScope.launch` (and the same for other builders). Thanks!